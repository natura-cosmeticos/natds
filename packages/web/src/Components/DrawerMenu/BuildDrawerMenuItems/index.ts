import BuildDrawerMenuItems from './BuildDrawerMenuItems'

export { BuildDrawerMenuItems }
export default BuildDrawerMenuItems
