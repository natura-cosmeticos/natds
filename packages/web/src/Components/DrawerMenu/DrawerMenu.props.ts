import * as React from 'react'

export interface IDrawerMenuProps {
  children?: React.ReactNode;
  component?: React.ElementType;
  list?: Array<unknown>;
}
