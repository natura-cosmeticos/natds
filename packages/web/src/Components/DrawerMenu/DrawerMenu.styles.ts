import { createStyles } from '../../styles/createStyles'
import { makeStyles } from '../../styles/makeStyles'
import { IThemeWeb } from '../../Themes'

const style = ({ sizes }: IThemeWeb) => createStyles({
  root: {
    flex: '1 1 auto'
  },
  listPadding: {
    marginBottom: sizes?.tiny,
    marginTop: sizes?.tiny,
    paddingBottom: sizes?.none,
    paddingTop: sizes?.none
  }
})

export const useStyles = makeStyles(style, { name: 'NatDSDrawerMenu' })
export default useStyles
