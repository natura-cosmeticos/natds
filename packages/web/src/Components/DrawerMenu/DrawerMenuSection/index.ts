import DrawerMenuSection from './DrawerMenuSection'

export { IDrawerMenuSectionProps } from './DrawerMenuSection.props'
export { DrawerMenuSection }
export default DrawerMenuSection
