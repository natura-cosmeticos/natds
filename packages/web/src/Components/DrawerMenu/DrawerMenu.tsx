// eslint-disable-next-line no-use-before-define
import * as React from 'react'
import { List } from '../List'
import { IDrawerMenuProps } from './DrawerMenu.props'
import { BuildDrawerMenuItems } from './BuildDrawerMenuItems'
import { useStyles } from './DrawerMenu.styles'

/**
 * ## Importing
 *
 * ```
 * import { DrawerMenu } from '@naturacosmeticos/natds-web';
 * ```
 */
export const DrawerMenu = React.forwardRef<HTMLElement, IDrawerMenuProps>((
  props: IDrawerMenuProps, ref
) => {
  const {
    children,
    component: Component = 'div',
    list,
    ...otherProps
  } = props

  const { root, listPadding } = useStyles()

  const content = children || (
  <List
    classes={{ padding: listPadding }}
    dense
  >
    {list && list.map((drawerMenuSectionProps, index) => (
      <BuildDrawerMenuItems {...drawerMenuSectionProps} key={index} />
    ))}
  </List>
  )

  return (
    <Component {...otherProps} className={root} ref={ref}>
      {content}
    </Component>
  )
})

DrawerMenu.displayName = 'DrawerMenu'

export default DrawerMenu
