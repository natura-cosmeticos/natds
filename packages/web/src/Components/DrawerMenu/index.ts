import DrawerMenu from './DrawerMenu'
import { DrawerMenuItem } from './DrawerMenuItem'
import { DrawerMenuSection } from './DrawerMenuSection'
import { MenuItemText } from './MenuItemText'
import { BuildDrawerMenuItems } from './BuildDrawerMenuItems'

export { IDrawerMenuProps } from './DrawerMenu.props'
export {
  DrawerMenu, DrawerMenuItem, DrawerMenuSection, MenuItemText, BuildDrawerMenuItems
}
export default DrawerMenu
