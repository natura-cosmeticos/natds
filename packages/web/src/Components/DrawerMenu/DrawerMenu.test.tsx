import * as React from 'react'
import * as TestRenderer from 'react-test-renderer'
import DrawerMenu from './DrawerMenu'
import { IDrawerMenuSectionProps } from './DrawerMenuSection'

const listBasic = [
  {
    icon: 'outlined-action-love',
    name: 'Menu Item'
  }
]

const listComplete = [
  {
    icon: 'outlined-action-love',
    name: 'Menu Item'
  },
  {
    badge: 'Novo',
    icon: 'outlined-action-love',
    name: 'Menu Item',
    onSelect: () => null
  },
  {
    badge: 'Novo',
    icon: 'outlined-action-love',
    list: [
      {
        name: 'Menu Item',
        icon: 'outlined-action-love'
      },
      {
        name: 'Menu Item',
        icon: 'outlined-action-love',
        badge: 'New'
      }
    ],
    name: 'Menu Item',
    onToggle: () => null,
    opened: false
  },
  {
    section: 'section'
  },
  {
    name: 'Menu Item',
    icon: 'outlined-action-love',
    badge: 'Novo'
  }
]

describe('DrawerMenu component', () => {
  test('should match snapshot - DrawerMenu list', () => {
    const component = TestRenderer.create(<DrawerMenu list={listComplete as unknown as [IDrawerMenuSectionProps]} />)

    expect(component).toMatchSnapshot()
  })

  test('should match snapshot - DrawerMenu custom component', () => {
    const component = TestRenderer.create(<DrawerMenu list={listComplete as unknown as [IDrawerMenuSectionProps]} component="section" />)

    expect(component).toMatchSnapshot()
  })

  test('should match snapshot - DrawerMenu custom children', () => {
    const children = <div>some custom children</div>
    const component = TestRenderer.create(<DrawerMenu list={listComplete as unknown as [IDrawerMenuSectionProps]}>{children}</DrawerMenu>)

    expect(component).toMatchSnapshot()
  })

  test('should render custom children', () => {
    const children = <div className="custom">some custom children</div>
    const component = TestRenderer.create(<DrawerMenu list={listBasic as [IDrawerMenuSectionProps]}>{children}</DrawerMenu>)

    expect(component).toMatchSnapshot()
  })

  test('should render list item', () => {
    const component = TestRenderer.create(<DrawerMenu list={listBasic as [IDrawerMenuSectionProps]} />)

    expect(component).toMatchSnapshot()
  })

  test('should render list item with badge', () => {
    const [listItem] = listBasic
    const list = [
      {
        ...listItem,
        badge: 'badge'
      }
    ]
    const component = TestRenderer.create(<DrawerMenu list={list as [IDrawerMenuSectionProps]} />)

    expect(component).toMatchSnapshot()
  })

  test('should render submenu item', () => {
    const component = TestRenderer.create(<DrawerMenu list={listComplete as unknown as [IDrawerMenuSectionProps]} />)

    expect(component).toMatchSnapshot()
  })

  test('should start with submenu opened', () => {
    const INDEX_TWO = 2
    const listItem = listComplete[INDEX_TWO]
    const list = [
      {
        ...listItem,
        opened: true
      }
    ]
    const component = TestRenderer.create(<DrawerMenu list={list as unknown as [IDrawerMenuSectionProps]} />)

    expect(component).toMatchSnapshot()
  })
})
