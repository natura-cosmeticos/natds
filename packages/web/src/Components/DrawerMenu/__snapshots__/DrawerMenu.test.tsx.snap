// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DrawerMenu component should match snapshot - DrawerMenu custom children 1`] = `
<div
  className="sc-dlfnbm gQlDxm"
>
  <div>
    some custom children
  </div>
</div>
`;

exports[`DrawerMenu component should match snapshot - DrawerMenu custom component 1`] = `
<section
  className="sc-dlfnbm gQlDxm"
>
  <ul
    className="MuiList-root MuiList-dense MuiList-padding makeStyles-padding-16"
  >
    <li
      aria-disabled={false}
      className="MuiButtonBase-root MuiListItem-root makeStyles-root-17 MuiListItem-dense MuiListItem-gutters MuiListItem-button"
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiListItemIcon-root"
      >
        <i
          aria-hidden={true}
          className="material-icons MuiIcon-root natds-icons natds-icons-outlined-action-love NatDSIcon-root-3 NatDSIcon-root-18"
        >
          outlined-action-love
        </i>
      </div>
      <div
        className="MuiListItemText-root MuiListItemText-dense"
      >
        <span
          className="MuiTypography-root MuiListItemText-primary MuiTypography-body2 MuiTypography-displayBlock"
        >
          Menu Item
        </span>
      </div>
      <span
        className="MuiTouchRipple-root"
      />
    </li>
    <li
      aria-disabled={false}
      className="MuiButtonBase-root MuiListItem-root makeStyles-root-19 MuiListItem-dense MuiListItem-gutters MuiListItem-button"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiListItemIcon-root"
      >
        <i
          aria-hidden={true}
          className="material-icons MuiIcon-root natds-icons natds-icons-outlined-action-love NatDSIcon-root-3 NatDSIcon-root-20"
        >
          outlined-action-love
        </i>
      </div>
      <div
        className="MuiListItemText-root MuiListItemText-dense"
      >
        <span
          className="MuiTypography-root MuiListItemText-primary MuiTypography-body2 MuiTypography-displayBlock"
        >
          <span
            className="MuiTypography-root MuiTypography-body2"
          >
            Menu Item
          </span>
          <span
            className="sc-gsTCUz gYcUew sc-bdfBwQ dQkjgA"
            data-type="standard"
            fontFamily="\\"Roboto\\", sans-serif"
            fontSize={12}
            fontWeight={400}
          >
            Novo
          </span>
        </span>
      </div>
      <span
        className="MuiTouchRipple-root"
      />
    </li>
    <li
      aria-disabled={false}
      className="MuiButtonBase-root MuiListItem-root makeStyles-root-22 MuiListItem-dense MuiListItem-gutters MuiListItem-button"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiListItemIcon-root makeStyles-root-24"
      >
        <i
          aria-hidden={true}
          className="material-icons MuiIcon-root natds-icons natds-icons-outlined-action-love NatDSIcon-root-3 NatDSIcon-root-25"
        >
          outlined-action-love
        </i>
      </div>
      <div
        className="MuiListItemText-root MuiListItemText-dense"
      >
        <span
          className="MuiTypography-root MuiListItemText-primary MuiTypography-body2 MuiTypography-displayBlock"
        >
          <span
            className="MuiTypography-root MuiTypography-body2"
          >
            Menu Item
          </span>
          <span
            className="sc-gsTCUz gYcUew sc-bdfBwQ dQkjgA"
            data-type="standard"
            fontFamily="\\"Roboto\\", sans-serif"
            fontSize={12}
            fontWeight={400}
          >
            Novo
          </span>
        </span>
      </div>
      <i
        aria-hidden={true}
        className="material-icons MuiIcon-root natds-icons natds-icons-outlined-navigation-arrowbottom NatDSIcon-root-3 NatDSIcon-root-26"
      >
        outlined-navigation-arrowbottom
      </i>
      <span
        className="MuiTouchRipple-root"
      />
    </li>
    <li
      className="MuiListSubheader-root MuiListSubheader-sticky MuiListSubheader-gutters"
    >
      <span
        className="MuiTypography-root MuiTypography-caption"
      >
        section
      </span>
    </li>
    <li
      aria-disabled={false}
      className="MuiButtonBase-root MuiListItem-root makeStyles-root-28 MuiListItem-dense MuiListItem-gutters MuiListItem-button"
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiListItemIcon-root"
      >
        <i
          aria-hidden={true}
          className="material-icons MuiIcon-root natds-icons natds-icons-outlined-action-love NatDSIcon-root-3 NatDSIcon-root-29"
        >
          outlined-action-love
        </i>
      </div>
      <div
        className="MuiListItemText-root MuiListItemText-dense"
      >
        <span
          className="MuiTypography-root MuiListItemText-primary MuiTypography-body2 MuiTypography-displayBlock"
        >
          <span
            className="MuiTypography-root MuiTypography-body2"
          >
            Menu Item
          </span>
          <span
            className="sc-gsTCUz gYcUew sc-bdfBwQ dQkjgA"
            data-type="standard"
            fontFamily="\\"Roboto\\", sans-serif"
            fontSize={12}
            fontWeight={400}
          >
            Novo
          </span>
        </span>
      </div>
      <span
        className="MuiTouchRipple-root"
      />
    </li>
  </ul>
</section>
`;

exports[`DrawerMenu component should match snapshot - DrawerMenu list 1`] = `
<div
  className="sc-dlfnbm gQlDxm"
>
  <ul
    className="MuiList-root MuiList-dense MuiList-padding makeStyles-padding-1"
  >
    <li
      aria-disabled={false}
      className="MuiButtonBase-root MuiListItem-root makeStyles-root-2 MuiListItem-dense MuiListItem-gutters MuiListItem-button"
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiListItemIcon-root"
      >
        <i
          aria-hidden={true}
          className="material-icons MuiIcon-root natds-icons natds-icons-outlined-action-love NatDSIcon-root-3 NatDSIcon-root-4"
        >
          outlined-action-love
        </i>
      </div>
      <div
        className="MuiListItemText-root MuiListItemText-dense"
      >
        <span
          className="MuiTypography-root MuiListItemText-primary MuiTypography-body2 MuiTypography-displayBlock"
        >
          Menu Item
        </span>
      </div>
    </li>
    <li
      aria-disabled={false}
      className="MuiButtonBase-root MuiListItem-root makeStyles-root-5 MuiListItem-dense MuiListItem-gutters MuiListItem-button"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiListItemIcon-root"
      >
        <i
          aria-hidden={true}
          className="material-icons MuiIcon-root natds-icons natds-icons-outlined-action-love NatDSIcon-root-3 NatDSIcon-root-6"
        >
          outlined-action-love
        </i>
      </div>
      <div
        className="MuiListItemText-root MuiListItemText-dense"
      >
        <span
          className="MuiTypography-root MuiListItemText-primary MuiTypography-body2 MuiTypography-displayBlock"
        >
          <span
            className="MuiTypography-root MuiTypography-body2"
          >
            Menu Item
          </span>
          <span
            className="sc-gsTCUz gYcUew sc-bdfBwQ dQkjgA"
            data-type="standard"
            fontFamily="\\"Roboto\\", sans-serif"
            fontSize={12}
            fontWeight={400}
          >
            Novo
          </span>
        </span>
      </div>
    </li>
    <li
      aria-disabled={false}
      className="MuiButtonBase-root MuiListItem-root makeStyles-root-8 MuiListItem-dense MuiListItem-gutters MuiListItem-button"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiListItemIcon-root makeStyles-root-10"
      >
        <i
          aria-hidden={true}
          className="material-icons MuiIcon-root natds-icons natds-icons-outlined-action-love NatDSIcon-root-3 NatDSIcon-root-11"
        >
          outlined-action-love
        </i>
      </div>
      <div
        className="MuiListItemText-root MuiListItemText-dense"
      >
        <span
          className="MuiTypography-root MuiListItemText-primary MuiTypography-body2 MuiTypography-displayBlock"
        >
          <span
            className="MuiTypography-root MuiTypography-body2"
          >
            Menu Item
          </span>
          <span
            className="sc-gsTCUz gYcUew sc-bdfBwQ dQkjgA"
            data-type="standard"
            fontFamily="\\"Roboto\\", sans-serif"
            fontSize={12}
            fontWeight={400}
          >
            Novo
          </span>
        </span>
      </div>
      <i
        aria-hidden={true}
        className="material-icons MuiIcon-root natds-icons natds-icons-outlined-navigation-arrowbottom NatDSIcon-root-3 NatDSIcon-root-12"
      >
        outlined-navigation-arrowbottom
      </i>
    </li>
    <li
      className="MuiListSubheader-root MuiListSubheader-sticky MuiListSubheader-gutters"
    >
      <span
        className="MuiTypography-root MuiTypography-caption"
      >
        section
      </span>
    </li>
    <li
      aria-disabled={false}
      className="MuiButtonBase-root MuiListItem-root makeStyles-root-14 MuiListItem-dense MuiListItem-gutters MuiListItem-button"
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiListItemIcon-root"
      >
        <i
          aria-hidden={true}
          className="material-icons MuiIcon-root natds-icons natds-icons-outlined-action-love NatDSIcon-root-3 NatDSIcon-root-15"
        >
          outlined-action-love
        </i>
      </div>
      <div
        className="MuiListItemText-root MuiListItemText-dense"
      >
        <span
          className="MuiTypography-root MuiListItemText-primary MuiTypography-body2 MuiTypography-displayBlock"
        >
          <span
            className="MuiTypography-root MuiTypography-body2"
          >
            Menu Item
          </span>
          <span
            className="sc-gsTCUz gYcUew sc-bdfBwQ dQkjgA"
            data-type="standard"
            fontFamily="\\"Roboto\\", sans-serif"
            fontSize={12}
            fontWeight={400}
          >
            Novo
          </span>
        </span>
      </div>
    </li>
  </ul>
</div>
`;

exports[`DrawerMenu component should render custom children 1`] = `
<div
  className="sc-dlfnbm gQlDxm"
>
  <div
    className="custom"
  >
    some custom children
  </div>
</div>
`;

exports[`DrawerMenu component should render list item 1`] = `
<div
  className="sc-dlfnbm gQlDxm"
>
  <ul
    className="MuiList-root MuiList-dense MuiList-padding makeStyles-padding-30"
  >
    <li
      aria-disabled={false}
      className="MuiButtonBase-root MuiListItem-root makeStyles-root-31 MuiListItem-dense MuiListItem-gutters MuiListItem-button"
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiListItemIcon-root"
      >
        <i
          aria-hidden={true}
          className="material-icons MuiIcon-root natds-icons natds-icons-outlined-action-love NatDSIcon-root-3 NatDSIcon-root-32"
        >
          outlined-action-love
        </i>
      </div>
      <div
        className="MuiListItemText-root MuiListItemText-dense"
      >
        <span
          className="MuiTypography-root MuiListItemText-primary MuiTypography-body2 MuiTypography-displayBlock"
        >
          Menu Item
        </span>
      </div>
    </li>
  </ul>
</div>
`;

exports[`DrawerMenu component should render list item with badge 1`] = `
<div
  className="sc-dlfnbm gQlDxm"
>
  <ul
    className="MuiList-root MuiList-dense MuiList-padding makeStyles-padding-33"
  >
    <li
      aria-disabled={false}
      className="MuiButtonBase-root MuiListItem-root makeStyles-root-34 MuiListItem-dense MuiListItem-gutters MuiListItem-button"
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiListItemIcon-root"
      >
        <i
          aria-hidden={true}
          className="material-icons MuiIcon-root natds-icons natds-icons-outlined-action-love NatDSIcon-root-3 NatDSIcon-root-35"
        >
          outlined-action-love
        </i>
      </div>
      <div
        className="MuiListItemText-root MuiListItemText-dense"
      >
        <span
          className="MuiTypography-root MuiListItemText-primary MuiTypography-body2 MuiTypography-displayBlock"
        >
          <span
            className="MuiTypography-root MuiTypography-body2"
          >
            Menu Item
          </span>
          <span
            className="sc-gsTCUz gYcUew sc-bdfBwQ dQkjgA"
            data-type="standard"
            fontFamily="\\"Roboto\\", sans-serif"
            fontSize={12}
            fontWeight={400}
          >
            badge
          </span>
        </span>
      </div>
      <span
        className="MuiTouchRipple-root"
      />
    </li>
  </ul>
</div>
`;

exports[`DrawerMenu component should render submenu item 1`] = `
<div
  className="sc-dlfnbm gQlDxm"
>
  <ul
    className="MuiList-root MuiList-dense MuiList-padding makeStyles-padding-36"
  >
    <li
      aria-disabled={false}
      className="MuiButtonBase-root MuiListItem-root makeStyles-root-37 MuiListItem-dense MuiListItem-gutters MuiListItem-button"
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiListItemIcon-root"
      >
        <i
          aria-hidden={true}
          className="material-icons MuiIcon-root natds-icons natds-icons-outlined-action-love NatDSIcon-root-3 NatDSIcon-root-38"
        >
          outlined-action-love
        </i>
      </div>
      <div
        className="MuiListItemText-root MuiListItemText-dense"
      >
        <span
          className="MuiTypography-root MuiListItemText-primary MuiTypography-body2 MuiTypography-displayBlock"
        >
          Menu Item
        </span>
      </div>
    </li>
    <li
      aria-disabled={false}
      className="MuiButtonBase-root MuiListItem-root makeStyles-root-39 MuiListItem-dense MuiListItem-gutters MuiListItem-button"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiListItemIcon-root"
      >
        <i
          aria-hidden={true}
          className="material-icons MuiIcon-root natds-icons natds-icons-outlined-action-love NatDSIcon-root-3 NatDSIcon-root-40"
        >
          outlined-action-love
        </i>
      </div>
      <div
        className="MuiListItemText-root MuiListItemText-dense"
      >
        <span
          className="MuiTypography-root MuiListItemText-primary MuiTypography-body2 MuiTypography-displayBlock"
        >
          <span
            className="MuiTypography-root MuiTypography-body2"
          >
            Menu Item
          </span>
          <span
            className="sc-gsTCUz gYcUew sc-bdfBwQ dQkjgA"
            data-type="standard"
            fontFamily="\\"Roboto\\", sans-serif"
            fontSize={12}
            fontWeight={400}
          >
            Novo
          </span>
        </span>
      </div>
    </li>
    <li
      aria-disabled={false}
      className="MuiButtonBase-root MuiListItem-root makeStyles-root-42 MuiListItem-dense MuiListItem-gutters MuiListItem-button"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiListItemIcon-root makeStyles-root-44"
      >
        <i
          aria-hidden={true}
          className="material-icons MuiIcon-root natds-icons natds-icons-outlined-action-love NatDSIcon-root-3 NatDSIcon-root-45"
        >
          outlined-action-love
        </i>
      </div>
      <div
        className="MuiListItemText-root MuiListItemText-dense"
      >
        <span
          className="MuiTypography-root MuiListItemText-primary MuiTypography-body2 MuiTypography-displayBlock"
        >
          <span
            className="MuiTypography-root MuiTypography-body2"
          >
            Menu Item
          </span>
          <span
            className="sc-gsTCUz gYcUew sc-bdfBwQ dQkjgA"
            data-type="standard"
            fontFamily="\\"Roboto\\", sans-serif"
            fontSize={12}
            fontWeight={400}
          >
            Novo
          </span>
        </span>
      </div>
      <i
        aria-hidden={true}
        className="material-icons MuiIcon-root natds-icons natds-icons-outlined-navigation-arrowbottom NatDSIcon-root-3 NatDSIcon-root-46"
      >
        outlined-navigation-arrowbottom
      </i>
    </li>
    <li
      className="MuiListSubheader-root MuiListSubheader-sticky MuiListSubheader-gutters"
    >
      <span
        className="MuiTypography-root MuiTypography-caption"
      >
        section
      </span>
    </li>
    <li
      aria-disabled={false}
      className="MuiButtonBase-root MuiListItem-root makeStyles-root-48 MuiListItem-dense MuiListItem-gutters MuiListItem-button"
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiListItemIcon-root"
      >
        <i
          aria-hidden={true}
          className="material-icons MuiIcon-root natds-icons natds-icons-outlined-action-love NatDSIcon-root-3 NatDSIcon-root-49"
        >
          outlined-action-love
        </i>
      </div>
      <div
        className="MuiListItemText-root MuiListItemText-dense"
      >
        <span
          className="MuiTypography-root MuiListItemText-primary MuiTypography-body2 MuiTypography-displayBlock"
        >
          <span
            className="MuiTypography-root MuiTypography-body2"
          >
            Menu Item
          </span>
          <span
            className="sc-gsTCUz gYcUew sc-bdfBwQ dQkjgA"
            data-type="standard"
            fontFamily="\\"Roboto\\", sans-serif"
            fontSize={12}
            fontWeight={400}
          >
            Novo
          </span>
        </span>
      </div>
    </li>
  </ul>
</div>
`;

exports[`DrawerMenu component should start with submenu opened 1`] = `
<div
  className="sc-dlfnbm gQlDxm"
>
  <ul
    className="MuiList-root MuiList-dense MuiList-padding makeStyles-padding-50"
  >
    <li
      aria-disabled={false}
      className="MuiButtonBase-root MuiListItem-root makeStyles-root-52 MuiListItem-dense MuiListItem-gutters MuiListItem-button Mui-selected makeStyles-selected-53"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiListItemIcon-root makeStyles-root-54"
      >
        <i
          aria-hidden={true}
          className="material-icons MuiIcon-root natds-icons natds-icons-outlined-action-love NatDSIcon-root-3 NatDSIcon-root-55"
        >
          outlined-action-love
        </i>
      </div>
      <div
        className="MuiListItemText-root MuiListItemText-dense"
      >
        <span
          className="MuiTypography-root MuiListItemText-primary MuiTypography-body2 MuiTypography-displayBlock"
        >
          <span
            className="MuiTypography-root MuiTypography-body2"
          >
            Menu Item
          </span>
          <span
            className="sc-gsTCUz gYcUew sc-bdfBwQ dQkjgA"
            data-type="standard"
            fontFamily="\\"Roboto\\", sans-serif"
            fontSize={12}
            fontWeight={400}
          >
            Novo
          </span>
        </span>
      </div>
      <i
        aria-hidden={true}
        className="material-icons MuiIcon-root natds-icons natds-icons-outlined-navigation-arrowtop NatDSIcon-root-3 NatDSIcon-root-56"
      >
        outlined-navigation-arrowtop
      </i>
      <span
        className="MuiTouchRipple-root"
      />
    </li>
    <li
      className="MuiCollapse-container MuiCollapse-entered"
      style={
        Object {
          "minHeight": "0px",
        }
      }
    >
      <div
        className="MuiCollapse-wrapper"
      >
        <div
          className="MuiCollapse-wrapperInner"
        >
          <ul
            className="MuiList-root MuiList-dense MuiList-padding makeStyles-padding-51"
          >
            <li
              aria-disabled={false}
              className="MuiButtonBase-root MuiListItem-root makeStyles-root-57 MuiListItem-dense MuiListItem-gutters MuiListItem-button"
              onBlur={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="button"
              tabIndex={0}
            >
              <div
                className="MuiListItemIcon-root"
              >
                <i
                  aria-hidden={true}
                  className="material-icons MuiIcon-root natds-icons natds-icons-outlined-action-love NatDSIcon-root-3 NatDSIcon-root-58"
                >
                  outlined-action-love
                </i>
              </div>
              <div
                className="MuiListItemText-root MuiListItemText-dense"
              >
                <span
                  className="MuiTypography-root MuiListItemText-primary MuiTypography-body2 MuiTypography-displayBlock"
                >
                  Menu Item
                </span>
              </div>
              <span
                className="MuiTouchRipple-root"
              />
            </li>
            <li
              aria-disabled={false}
              className="MuiButtonBase-root MuiListItem-root makeStyles-root-59 MuiListItem-dense MuiListItem-gutters MuiListItem-button"
              onBlur={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="button"
              tabIndex={0}
            >
              <div
                className="MuiListItemIcon-root"
              >
                <i
                  aria-hidden={true}
                  className="material-icons MuiIcon-root natds-icons natds-icons-outlined-action-love NatDSIcon-root-3 NatDSIcon-root-60"
                >
                  outlined-action-love
                </i>
              </div>
              <div
                className="MuiListItemText-root MuiListItemText-dense"
              >
                <span
                  className="MuiTypography-root MuiListItemText-primary MuiTypography-body2 MuiTypography-displayBlock"
                >
                  <span
                    className="MuiTypography-root MuiTypography-body2"
                  >
                    Menu Item
                  </span>
                  <span
                    className="sc-gsTCUz gYcUew sc-bdfBwQ dQkjgA"
                    data-type="standard"
                    fontFamily="\\"Roboto\\", sans-serif"
                    fontSize={12}
                    fontWeight={400}
                  >
                    New
                  </span>
                </span>
              </div>
              <span
                className="MuiTouchRipple-root"
              />
            </li>
          </ul>
        </div>
      </div>
    </li>
  </ul>
</div>
`;
